apps/dialtone-documentation/docs/.vuepress/common/token-utilities.js

```diff
@@ -45,3 +46,138 @@ export function getComposedShadowTokens (theme) {
     });
   return tokens;
 }
+
+// eslint-disable-next-line complexity
+export const addTokenToStructure = (token, format, structure) => {
```

Comment:
Seems like this function can be splitted in 3.
Would be cool to separate some code from it to have it more clear.

-----
```
const [key, value] = token;
  if (!value[FORMAT_MAP[format]] || !value[FORMAT_MAP.CSS]) return;
```
-----
```
const { name, value: tokenValue, description, keywords } = value[FORMAT_MAP[format]];
  const { value: exampleValue, name: exampleName } = value[FORMAT_MAP.CSS];
  const displayToken = { exampleValue, exampleName, name, tokenValue, description, keywords };
```
-----
```
const splitKeys = key.split('/');

  // COLOR
  if (key.startsWith('color') && SUBCATEGORY_MAP.color.includes(splitKeys[1])) {
....
```

Just to follow in a better way what we are doing here and what the function should build.


About the `ifs`, seems like there are some pattern to follow.
What we have:

1. color
2. color/brand
3. opacity
4. color
5. typography
6. font
7. font/text-case
8. font/lineHeight
9. CATEGORY_MAP.component.includes(splitKeys[0]) && (splitKeys[1] === '`font`' || splitKeys[1] === 'lineHeight')
10. shadow
11. size
12. size/radius
13. size/border
14. size
15. `if (CATEGORY_MAP.component.includes(splitKeys[0]) && splitKeys[1] === '`size`') { if (splitKeys[2] === 'border') {`
16. space && negative
17. space && percent
18. space


Ok, its a lot, and also it depends on the order.

We definetly should separet these in different function, but these functions can be less specific.
As I see we can have something like:

1. color
2. opacity
3. typography
4. font
5. shadow
6. size
7. space 

It can be sth like:
`handleColorTokens(param_1, param_2, param_3)`.

Reply:
About the splitting, it's fixed.
About adding `handleColorTokens(params)`, `handleTypographyTokens(params)`, etc. I don't think it adds much readability, and also it might add complexity because the function would have to do two things:
1- Add the tokens to the structure
2- Return if it handled the tokens, or if not, in that case we should continue looking for other categories